pkg/controller/infinispan/resources/operatorconfig/operatorconfig_controller.go

```diff
@@ -0,0 +1,135 @@
+	}
+	operatorConfigPredicate := predicate.Funcs{
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return e.Meta.GetName() == configMapName && e.Meta.GetNamespace() == operatorNS
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return e.Meta.GetName() == configMapName && e.Meta.GetNamespace() == operatorNS
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return e.MetaNew.GetName() == configMapName && e.MetaNew.GetNamespace() == operatorNS
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return e.Meta.GetName() == configMapName && e.Meta.GetNamespace() == operatorNS
+		},
+	}
+	// Watch for changes to global operator config
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, operatorConfigPredicate)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// newOperatorConfigReconciler returns a new reconcile.Reconciler
+func newOperatorConfigReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileOperatorConfig{client: mgr.GetClient(), scheme: mgr.GetScheme(), Log: logf.Log.WithName("operator-config-controller"), Kube: kube.NewKubernetesFromController(mgr)}
+}
+
+var _ reconcile.Reconciler = &ReconcileOperatorConfig{}
+
+// Reconcile reads the operator configuration in the `infinispan-operator-config` ConfigMap and
+// applies the configuration to the cluster.
+func (r *ReconcileOperatorConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	operatorNs, err := getOperatorNamespace()
+	isRunningLocal := false
+	if err == k8sutil.ErrRunLocal {
+		isRunningLocal = true
+	}
+	if err != nil && !isRunningLocal {
+		r.Log.Error(err, "Error getting operator runtime namespace")
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	configMap := &corev1.ConfigMap{}
+	config := getDefaultOperatorConfig()
+	err = r.client.Get(ctx, types.NamespacedName{Namespace: operatorNs, Name: configMapName}, configMap)
+	if err != nil && !errors.IsNotFound(err) {
+		r.Log.Error(err, "Error getting operator configuration resource")
+		return reconcile.Result{Requeue: true}, nil
+	}
+	if err == nil {
+		config = configMap.Data
+	}
+	// Reconcile grafana
+	res, err := r.reconcileGrafana(config, operatorNs)
+	return *res, err
+}
+
+// getOperatorNamespace returns the operator namespace or the WATCH_NAMESPACE
+// value if the operator is running outside the cluster.
+// "OSDK_FORCE_RUN_MODE"="local" must be set if running locally (not in the cluster)
+func getOperatorNamespace() (string, error) {
+	operatorNs, err := k8sutil.GetOperatorNamespace()
+	// This makes everything works even running outside the cluster
+	if err != nil && err == k8sutil.ErrRunLocal {
+		operatorNs, err = k8sutil.GetWatchNamespace()
+		if err != nil {
+			return "", err
+		}
+		if operatorNs == "" {
+			return "", nil
+		}
+		return strings.Split(operatorNs, ",")[0], nil
+	}
+	return operatorNs, err
+}
+
+func getDefaultOperatorConfig() map[string]string {
+	return map[string]string{}
```

Comment:
Should we just make this a var?

Reply:
let's design this when we actually need default config

Reply:
populated with valued and moved to var